sdv/metadata.py

```diff
@@ -116,6 +116,76 @@ def _analyze_relationships(self):
                         self._child_map[parent].add(table)
                         self._parent_map[table].add(parent)
 
+    def _validate_dataset_structure(self):
+        """Make sure that all the tables have at most one parent."""
+        for table in self.get_tables():
+            if len(self.get_parents(table)) > 1:
+                raise ValueError('Some tables have multiple parents, which is not supported yet.')
+
+    def _validate_table(self, table_name, table_meta, tables=None):
+        """Validate metadata table.
+
+        Args:
+            table_name (str):
+                Table name to be validated.
+            table_meta (dict):
+                Table metadata to be validated.
+            tables (dict):
+                Dictionary with the name of the table and their data.
+                If it's provided, assert metadata types are valid and
+                metadata fields exist in data columns.
+        """
+        dtypes = self.get_dtypes(table_name, ids=True)
+
+        # Primary key field exists and its type is 'id'
+        primary_key = table_meta.get('primary_key')
+        if primary_key:
+            pk_field = table_meta['fields'].get(primary_key)
+
+            if not pk_field:
+                raise MetadataError('Primary key is not an existing field.')
+
+            if pk_field['type'] != 'id':
+                raise MetadataError('Primary key is not of type `id`.')
+
+        if tables:
+            data = tables[table_name]
+
+            for column in data:
+                data[column].astype(dtypes[column])
+
+            # assert all dtypes are in data
+            for field in dtypes:
```

Comment:
Instead of looping a second time, during the first loop `pop` the values out of the `dtypes` dict.
When you reach this point, `dtypes` should be empty, so you can use:

```
if dtypes:
    raise MetadataError(
        'Missing columns on table {}: {}'.format(
            table_name, list(dtypes.keys()))
    )
```